pilot/pkg/model/context.go

```diff
@@ -819,20 +819,6 @@ func (node *Proxy) SetServiceInstances(serviceDiscovery ServiceDiscovery) {
 	node.ServiceInstances = instances
 }
 
-// SetWorkloadLabels will set the node.Metadata.Labels only when it is nil.
-func (node *Proxy) SetWorkloadLabels(env *Environment) {
-	// First get the workload labels from node meta
-	if len(node.Metadata.Labels) > 0 {
-		return
-	}
-
-	// Fallback to calling GetProxyWorkloadLabels
-	l := env.GetProxyWorkloadLabels(node)
```

Comment:
so do we never call GetProxyWorkloadLabels anymore?

Reply:
yes. We always expect and use labels from Metadata only.

Reply:
If the function is never used then seems like we should clean it up entirely ?

Reply:
Yes. I was thinking of that. Since it is in service discovery interface - involves lot more changes. Will do it in a follow-up